controller/src/beerocks/cli/beerocks_cli_bml.cpp

```diff
@@ -1806,5 +1913,63 @@ int cli_bml::get_dcs_scan_results(const std::string &radio_mac, unsigned int max
 {
     std::cout << "get_dcs_scan_results" << std::endl;
 
+    if (max_results_size == 0) {
+        std::cout << __func__ << "invalid input, max_results_size==0" << std::endl;
+        return -1;
+    }
+
+    struct BML_DCS_NEIGHBOR_AP *results = nullptr;
+
+    results =
+        (struct BML_DCS_NEIGHBOR_AP *)calloc(max_results_size, sizeof(struct BML_DCS_NEIGHBOR_AP));
```

Comment:
There is absolutely no good reason to use `calloc` or `alloc` in c++, and especially in this function where you simply print the returned results and free them right after.
Simply declare the results on the stack:

`BML_DCS_NEIGHBOR_AP results[max_results_size];`

Also - why is the `max_results_size` argument even needed? This can also be statically declared in the function.

Reply:
the max_results_size is to simulate the max-results-size limit also used in <wave-api> in RDKB (currently set as 200)
when DCS supports ordering of the results by RSSI or another ranking, it is possible to the get only the X best results even if more are available

I replaced the <calloc> with a stack declaration.

Reply:
The best would be to change the DCS WLAN APIs as well - is it feasible?

Reply:
it might be - you can create a JIRA to check this at a later time - shouldn't block this PR